gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -253,6 +265,19 @@ public void launchJob(JobListener jobListener)
       runWorkUnits(workUnits.get());
       Instrumented.endTimer(jobRunTimer);
 
+      switch (jobState.getState()) {
```

Comment:
You can actually do `this.eventSubmitter.submit(CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.UPPER_CAMEL, "JOB_" + jobState.getState()))` to replace the `switch` clause;